### PR TITLE
Swap docs to MkDocs

### PR DESCRIPTION
**Pull request by schlegelp**
This PR switches from Sphinx to MkDocs Material. A preview is already visible here: https://navis-org.github.io/navis/

Pretty happy with the results but there were a couple snags along the way:

1. Tutorials:
    With Sphinx we used NB convert to generated reST documents from the Jupyter notebooks. Here we switched to [mkdocs-gallery](https://smarie.github.io/mkdocs-gallery/) - a port of Sphinx gallery - which uses `.py` Python scripts (with the `# %%` cell syntax). Disadvantage: to get cell outputs (such as figures) we need to actually execute the script during the build. The issue with this is that (a) it takes a while (not too bad though) and (b) some example that e.g. need to download data or have a long run time (like the NBLASTs) had to be made static. Advantage: smaller file size and the build process effectively doubles as tests. `mkdocs-gallery` works nice for the actual tutorials. For rendering other documents such as the `quickstart.md` I used `markdown-exec`.
2. Autodocs 
    Sphinx had the nice autodoc functions (e.g. the `:: autosummary` directive) which automatically generate the html files for the docstrings. With MkDocs we have to do a bit more leg work. I ended up using `mkdocs-gen-files` but I had to write a custom script to collect the functions (see `gen_ref_pages2.py`). In addition, I used `mkdocs-macros-plugin` to write macro that lets me write e.g. `{{ autosummary("navis.BaseNeuron.bbox") }}` (see `main.py`) to replace the Sphinx `::autosummary` directive.

Additional changes:
- `TreeNeuron.soma_pos` can now be set to `None` to remove any soma
- `navis.Volume` now has a `.units` property similar to neurons 
- removed `neuprint.fetch_skeleton` from navis' neuprint interface to avoid confusion with our own `fetch_skeletons` (plural!)
- removed `navis.screenshot()` (may bring that back later) - use `Viewer.screenshot` instead
- implement `NeuronList.__or__` magic method
- new method: `NeuronList.get_neuron_attributes`
- new function: `navis.graph.skeleton_adjacency_matrix()`
- `Neuron.pre/postsynapses` is now case-insensitive when looking for the respective connector`type`
- plot2d/3d: `alpha` can now be given as a list/array with an alpha value for each neuron
- fixed the MICrONs interface includes new functions and improvements
- fixed a bug with `TreeNeuron.simple` where a soma could potentially lead to a non-branch, non-leaf node be retained
- fixed a couple issues in plot2d and pl3d

TODOs:
- [ ] add some missing gallery thumbnails
- [x] fix the ecosystem graphic
- [x] use `markdown-exec` to add some plots to quickstart (see [this](https://pawamoy.github.io/markdown-exec/gallery/#with-matplotlib) example; use `# markdown-exec: hide` to hide lines of code)
- [x] add tutorial for skeleton plotting
- [x] installation: move optional dependencies into a separate document?
- [ ] add some eyecandy to landing page?
- [x] remove TOC from landing page?
- [x] edit the card icons on landing page
- [ ] remove TOC from tutorials (should make them wider):
   ```
   ---
   hide:
     - navigation
     - toc
   ---
   ```
   This will require a fix in `mkdocs-gallery` though (see smarie/mkdocs-gallery#96)

### CODE REVIEW

**Comment by bdpedigo** [comment]
new docs look awesome! 

One thought - could be helpful to remove the circled extra dropdowns 

Screenshot 2024-09-03 at 4 26 07 PM

**Comment by bdpedigo** [comment]
we also had a lot of debate about this part but decided to also use top-level tabs as well to reduce the load on the side menu - seemed a matter of personal preference, though 

Screenshot 2024-09-03 at 4 28 10 PM

**Comment by schlegelp** [comment]
> One thought - could be helpful to remove the circled extra dropdowns

Yeah that's a bit of an eyesore. My hope is that the people will mainly look at/search the API documentation main page which is manually curated and should cover most top-level functions. The stuff under `navis` is automatically generated during the build using `mkdocs-gen-files` and `literate-nav` following [this recipe](https://mkdocstrings.github.io/recipes/#automatic-code-reference-pages) from `mkdocstrings`.

Basically, I didn't want to have to manually generate markdown files and populate them with the `::: class/function/module` syntax - `navis` is just too big for that.

Will see if I can somehow improve it.

**Comment by schlegelp** [comment]
> we also had a lot of debate about this part but decided to also use top-level tabs as well to reduce the load on the side menu - seemed a matter of personal preference, though

I experimented with this a bit and decided to go with the top-level tabs too. The main selling point for me was that I can deactivate the navigation and/or the toc for pages where they are not needed which significantly increases the available width for the actual content. The biggest impact would be on the tutorials but `mkdocs-gallery` does not (yet) pass through page headers (which you use to selectively deactivate navigation/toc).

**Comment by bdpedigo** [comment]
> > One thought - could be helpful to remove the circled extra dropdowns
> 
> Yeah that's a bit of an eyesore. My hope is that the people will mainly look at/search the API documentation main page which is manually curated and should cover most top-level functions. The stuff under `navis` is automatically generated during the build using `mkdocs-gen-files` and `literate-nav` following [this recipe](https://mkdocstrings.github.io/recipes/#automatic-code-reference-pages) from `mkdocstrings`.
> 
> Basically, I didn't want to have to manually generate markdown files and populate them with the `::: class/function/module` syntax - `navis` is just too big for that.
> 
> Will see if I can somehow improve it.

that all makes sense - its definitely not a huge deal either. let me see if I can find how I handled this in the past

**Comment by schlegelp** [comment]
Old and new tests are finally passing and I ticked off most TODOs.

The workflow for building & deploying docs hasn't actually run yet because it is only triggered when pushing to the main branch (which makes sense, I guess). I'm guessing I forgot to add some of the `mkdocs` plugins to the requirements but I will cross that bridge when we get there.

I think at this point I will go ahead and merge.
